legacy/builder/create_cmake_rule.go

```diff
@@ -17,30 +17,48 @@ package builder
 
 import (
 	"fmt"
-	"io/ioutil"
-	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
 
 	properties "github.com/arduino/go-properties-orderedmap"
 
+	"github.com/arduino/arduino-cli/arduino/globals"
 	"github.com/arduino/arduino-cli/legacy/builder/builder_utils"
 	"github.com/arduino/arduino-cli/legacy/builder/constants"
 	"github.com/arduino/arduino-cli/legacy/builder/types"
 	"github.com/arduino/arduino-cli/legacy/builder/utils"
 )
 
-var VALID_EXPORT_EXTENSIONS = map[string]bool{".h": true, ".c": true, ".hpp": true, ".hh": true, ".cpp": true, ".S": true, ".a": true, ".properties": true}
-var DOTHEXTENSION = map[string]bool{".h": true, ".hh": true, ".hpp": true}
-var DOTAEXTENSION = map[string]bool{".a": true}
+func stringArrayToLookupFunction(in []string) func(string) bool {
+	out := map[string]bool{}
+	for _, i := range in {
+		out[i] = true
+	}
+	return func(s string) bool {
+		return out[s]
+	}
+}
```

Comment:
I'd remove this in favor of changing `utils.CopyDir` to be similar to `utils.FindFilesInFolder`.